Tests/test_kwarg.py

```diff
@@ -152,614 +136,562 @@ def __new__(cls, a, *param, **kw):
+        self.CheckArgDictInit(v, 'abc', None, {'d':'def', 'e':'fgi', 'f':'ijk'})
+        self.assertEqual(type(v), NewInitKwAndExtraParam)
+        
+        v = NewInitKwAndExtraParam(a='abc')
+        self.CheckArgDictInit(v, 'abc', None, {})
+        self.assertEqual(type(v), NewInitKwAndExtraParam)
+        
+        v = NewInitKwAndExtraParam(a='abc',d='def')
+        self.CheckArgDictInit(v, 'abc', None, {'d':'def'})
+        self.assertEqual(type(v), NewInitKwAndExtraParam)
+        
+        v = NewInitKwAndExtraParam(a='abc',d='def', e='fgi')
+        self.CheckArgDictInit(v, 'abc', None, {'d':'def', 'e':'fgi'})
+        self.assertEqual(type(v), NewInitKwAndExtraParam)
+        
+        v = NewInitKwAndExtraParam(a='abc', d='def', e='fgi', f='ijk')
+        self.CheckArgDictInit(v, 'abc', None, {'d':'def', 'e':'fgi', 'f':'ijk'})
+        self.assertEqual(type(v), NewInitKwAndExtraParam)
+
+    def test_NewInitKwAndExtraParamAndParams(self):
+        v = NewInitKwAndExtraParamAndParams('abc')
+        self.CheckArgDict(NewInitKwAndExtraParamAndParams, 'abc', (), {})
+        self.assertEqual(type(v), NewInitKwAndExtraParamAndParams)
+        
+        v = NewInitKwAndExtraParamAndParams('abc', 'cde')
+        self.CheckArgDict(NewInitKwAndExtraParamAndParams, 'abc', ('cde',), {})
+        self.assertEqual(type(v), NewInitKwAndExtraParamAndParams)
+        
+        v = NewInitKwAndExtraParamAndParams('abc', 'cde', 'def')
+        self.CheckArgDict(NewInitKwAndExtraParamAndParams, 'abc', ('cde','def'), {})
+        self.assertEqual(type(v), NewInitKwAndExtraParamAndParams)
+        
+        v = NewKwAndExtraParamAndParams(a='abc', b='cde', e='def')
+        self.CheckArgDict(NewKwAndExtraParamAndParams, 'abc', (), {'b':'cde', 'e':'def'})
+        self.assertEqual(type(v), NewKwAndExtraParamAndParams)
+        
+        v = NewKwAndExtraParamAndParams('abc', 'cde', e='def', d='ghi')
+        self.CheckArgDict(NewKwAndExtraParamAndParams, 'abc', ('cde',), {'e':'def', 'd':'ghi'})
+        self.assertEqual(type(v), NewKwAndExtraParamAndParams)
+        
+        v = NewKwAndExtraParamAndParams('abc', e='def', f='ghi')
+        self.CheckArgDict(NewKwAndExtraParamAndParams, 'abc', (), {'e':'def', 'f':'ghi'})
+        self.assertEqual(type(v), NewKwAndExtraParamAndParams)
+
+    def test_list_derive(self):
+        """verify we can derive from list  & new up w/ keyword args"""
+        class ListSubcls(list):
+            def __new__(cls, **kw):
+                pass
+                
+        a = ListSubcls(a='abc')
+
+    def test_negTestFunc_testFunc_pw_kw_dupArg(self):
+        self.assertRaises(TypeError, testFunc_pw_kw, '234', a='234')
+
+    def test_negTestFunc_testFunc_kw_dupArg(self):
+        self.assertRaises(TypeError, testFunc_kw, '234', a='234')
+
+    def test_negTestFunc_ObjectSubClass_testFunc_pw_kw_dupArg(self):
+        o = ObjectSubClass()
+        try: o.testFunc_pw_kw(a='abc')
```

Comment:
I think it may have something to do with how the parameter is passed around (in the case of NOT using assertRaises, it is setting the value of the 'a' parameter, but when passed into assertRaises it is passed as **kwArgs. A comment is a good idea